From 585b9d311e97a6088b57a6c3716280063f8e5cf3 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 20 Jan 2018 21:17:27 +0100
Subject: openH264: enable ARM/ARM64 optimizations

Enable the optimizations not only for ChromeOS but for all compatbile ARM/ARM64 architectures
Limit threads auto-detect only for iOS

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
Signed-off-by: Saalim Quadri <danascape@gmail.com>
---
 third_party/openh264/BUILD.gn | 24 ++++++++++++++++++------
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/third_party/openh264/BUILD.gn b/third_party/openh264/BUILD.gn
index 287d71b2ccda3..212e2a6af987b 100644
--- a/third_party/openh264/BUILD.gn
+++ b/third_party/openh264/BUILD.gn
@@ -136,8 +136,12 @@ if (use_assembler) {
 
 source_set("common") {
   sources = openh264_common_sources
-  if (is_chromeos && current_cpu == "arm") {
-    sources += openh264_common_sources_asm_arm
+  if (is_chromeos || is_android) {
+    if (current_cpu == "arm") {
+      sources += openh264_common_sources_asm_arm
+    } else if (current_cpu == "arm64") {
+      sources += openh264_common_sources_asm_arm64
+    }
   }
   include_dirs = openh264_common_include_dirs
 
@@ -162,8 +166,12 @@ source_set("common") {
 
 source_set("processing") {
   sources = openh264_processing_sources
-  if (is_chromeos && current_cpu == "arm") {
-    sources += openh264_processing_sources_asm_arm
+  if (is_chromeos || is_android) {
+    if (current_cpu == "arm") {
+      sources += openh264_processing_sources_asm_arm
+    } else if (current_cpu == "arm64") {
+      sources += openh264_processing_sources_asm_arm64
+    }
   }
   include_dirs = openh264_processing_include_dirs
 
@@ -179,8 +187,12 @@ source_set("processing") {
 
 source_set("encoder") {
   sources = openh264_encoder_sources
-  if (is_chromeos && current_cpu == "arm") {
-    sources += openh264_encoder_sources_asm_arm
+  if (is_chromeos || is_android) {
+    if (current_cpu == "arm") {
+      sources += openh264_encoder_sources_asm_arm
+    } else if (current_cpu == "arm64") {
+      sources += openh264_encoder_sources_asm_arm64
+    }
   }
   include_dirs = openh264_encoder_include_dirs
 
-- 
2.55.0

